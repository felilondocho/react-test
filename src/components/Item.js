import React from 'react';
import injectSheet from 'react-jss'

const styles = {
  item: {
    backgroundColor: 'white',
    marginBottom: '20px',
    boxShadow: '0px 0px 4px 4px rgba(241,239,243,1)',
    padding: '20px',
    textAlign: 'left',
    position: 'relative',
    '&:after': {
      content: '""',
      width: '100%',
      height: '100%',
      position: 'absolute',
      top: '0',
      left: '0',
      boxShadow: '0px 0px 4px 4px rgba(0,0,0,1)',
      opacity: '0',
      transition: 'opacity 0.3s ease-in-out'
    },
    '&:hover:after': {
      opacity: '1'
    },
  },
  itemTitle: {
    margin: 0,
    marginBottom: '15px',
  },
  itemDescription: {
    margin: 0
  }
}

const Item = ({props, classes, itemElements}) => {
  return (
    <div className={classes.item}>
      <h4 className={classes.itemTitle}>{itemElements.ItemTitle}</h4>
      <p className={classes.itemDescription}>{itemElements.Description}</p>
    </div>
  );
}

export default injectSheet(styles)(Item);