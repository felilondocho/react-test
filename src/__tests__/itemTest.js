import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Item from '../components/Item';

Enzyme.configure({ adapter: new Adapter() });

describe('Item', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Item />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});