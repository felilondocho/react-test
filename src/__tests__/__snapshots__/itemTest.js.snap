// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<Item
  classes={
    Object {
      "item": "Item-item-0-1-1",
      "itemDescription": "Item-itemDescription-0-1-3",
      "itemTitle": "Item-itemTitle-0-1-2",
    }
  }
/>
`;
